### PR TITLE
Optimize and Clean up WalletRpcTest

### PR DESCRIPTION
**Pull request by benthecarman**
`be able to import an address` would sometimes fail because the receiver would not have seen the block yet, now if it hasn't seen it we will publish the block from that client as well.

### CODE REVIEW

File: bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/common/WalletRpcTest.scala [hunk 1]
```diff
@@ -357,9 +357,13 @@ class WalletRpcTest extends BitcoindRpcTest {
   }
 
   it should "be able to import an address" in {
+
+    val address = Bech32Address
```
Comment (Christewart) [inline bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/common/WalletRpcTest.scala:361]: Why have this hard coded?

Comment (benthecarman) [reply bitcoind-rpc-test/src/test/scala/org/bitcoins/rpc/common/WalletRpcTest.scala:361]: So we don't need to fetch an address from one of the clients, one less thing that can fail

File: testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala [hunk 2]
```diff
@@ -852,7 +865,14 @@ trait BitcoindRpcTestUtil extends BitcoinSLogger {
       txid <- fundMemPoolTransaction(sender, address, amount)
       addr <- sender.getNewAddress
       blockHash <- sender.generateToAddress(1, addr).map(_.head)
-      _ <- hasSeenBlock(receiver, blockHash)
+      seenBlock <- hasSeenBlock(receiver, blockHash)
+      _ <- if (seenBlock) {
+        FutureUtil.unit
+      } else {
+        sender
+          .getBlockRaw(blockHash)
```
Comment (Christewart) [inline testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala:873]: How does this work compared to `getBlock()` which in turn is querying `bitcoind`? Why will `getBlockRaw` succeed while `getBlock()` doesn't?

Comment (benthecarman) [reply testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala:873]: `getBlock` returns information about the block (a `GetBlockResult`) vs `getBlockRaw` returns an actual `Block`

Comment (Christewart) [reply testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala:873]: Why would `getBlockRaw` work and `getBlock` wouldn't work in the same bitcoind instance

Comment (benthecarman) [reply testkit/src/main/scala/org/bitcoins/testkit/rpc/BitcoindRpcTestUtil.scala:873]: The problem wasn't that they were failing, the problem was that the `receiver` had not seen the block, so we use `getBlockRaw` on the `sender` and give it to the `receiver` manually

**Pull request merged**